tests/foreman/api/test_discoveryrule.py

```diff
@@ -193,5 +193,7 @@ def test_positive_multi_provision_with_rule_limit(
 
     # Delete discovery rule
     rule.delete()
+    module_target_sat.api.Host(id=discovered_host1['id']).delete()
+    module_target_sat.api.DiscoveredHost(id=discovered_host2['id']).delete()
     with pytest.raises(HTTPError):
         rule.read()
```

Comment:
can this be added in request.addfinalizer?

Reply:
Added